src/Hl7.Fhir.Base/Serialization/engine/PocoSerializationEngine.cs

```diff
@@ -10,67 +10,58 @@
-        /// <param name="inspector">Reflection data of the POCO model to use.</param>
-        /// <param name="ignoreFilter">Predicate specifying which exceptions to ignore</param>
-        /// <param name="jsonSettings">Settings for json deserializing</param>
-        /// <param name="xmlSettings">Settings for xml deserializing</param>
-        public PocoSerializationEngine(ModelInspector inspector, Predicate<CodedException>? ignoreFilter=null, FhirJsonPocoDeserializerSettings? jsonSettings=null, FhirXmlPocoDeserializerSettings? xmlSettings=null)
+        
+        internal PocoSerializationEngine(ModelInspector inspector, Predicate<CodedException>? ignoreFilter = null, FhirJsonPocoDeserializerSettings? jsonDeserializerSettings = null, FhirJsonPocoSerializerSettings? jsonSerializerSettings = null, FhirXmlPocoDeserializerSettings? xmlSettings = null)
         {
             _inspector = inspector;
             _ignoreFilter = ignoreFilter ?? (_ => false);
-            _jsonSettings = jsonSettings ?? new FhirJsonPocoDeserializerSettings();
+            _jsonDeserializerSettings = jsonDeserializerSettings ?? new FhirJsonPocoDeserializerSettings();
+            _jsonSerializerSettings = jsonSerializerSettings ?? new FhirJsonPocoSerializerSettings();
             _xmlSettings = xmlSettings ?? new FhirXmlPocoDeserializerSettings();
         }
 
-        /// <inheritdoc />
-        public Resource DeserializeFromXml(string data)
+        internal PocoSerializationEngine(BaseFhirJsonPocoDeserializer deserializer,
+            BaseFhirJsonPocoSerializer serializer)
         {
-            var deserializer = new BaseFhirXmlPocoDeserializer(_inspector, _xmlSettings);
-            return deserializeAndIgnoreErrors(deserializer.TryDeserializeResource, data);
+            _jsonDeserializer = deserializer;
+            _jsonSerializer = serializer;
+            // dirty, but this constructor is really not supposed to be supported for much longer
+            var inspectorfield =
+                typeof(BaseFhirJsonPocoDeserializer).GetField("_inspector", BindingFlags.NonPublic | BindingFlags.Instance); 
+            _inspector = (inspectorfield!.GetValue(_jsonDeserializer) as ModelInspector)!;
+            _ignoreFilter = _ => false;
+            _xmlSettings = new FhirXmlPocoDeserializerSettings();
         }
-
-        /// <inheritdoc />
-        public Resource DeserializeFromJson(string data)
+        
+        private Base deserializeAndFilterErrors(TryDeserializer deserializer)
         {
-            var deserializer = new BaseFhirJsonPocoDeserializer(_inspector, _jsonSettings);
-            return deserializeAndIgnoreErrors(deserializer.TryDeserializeResource, data);
-        }
-
-        private Resource deserializeAndIgnoreErrors(TryDeserializer deserializer, string data)
-        {
-            bool success = deserializer(data, out var instance, out var issues);
+            (Base? instance, IEnumerable<CodedException> issues) = deserializer();
```

Comment:
You don't have to mention the types (but you can if you think it's clearer), `var (instance, issues) = deserializer()` would work too.